.github/workflows/docker.yaml

```diff
@@ -48,20 +54,53 @@ jobs:
         with:
           context: .
           file: "./docker/Dockerfile"
+          load: true
           push: true
           tags: |
-            mrcide/hintr:${{steps.extract_branch.outputs.branch}}
-            ghcr.io/${{github.repository}}:${{steps.extract_branch.outputs.branch}}
+            ${{env.TAG_SERVER_DH}}:${{steps.extract_branch.outputs.branch}}
+            ghcr.io/${{env.TAG_SERVER_GHCR}}:${{steps.extract_branch.outputs.branch}}
 
       - name: Build and push prod containers
+        uses: docker/build-push-action@v5
+        if: github.event_name != 'pull_request'
+        with:
+          context: .
+          file: "./docker/Dockerfile"
+          load: true
+          push: true
+          tags: |
+            ${{env.TAG_SERVER_DH}}:latest
+            ${{env.TAG_SERVER_DH}}:${{steps.extract_branch.outputs.branch}}
```

Comment:
do you want to port over the way im pulling the branch name from here https://github.com/mrc-ide/hint/pull/961/files#diff-8c20180da1290d5fde5f53c9e31e44c8764e13fc7280d99510f7b9e8875b9470R18-R28 otherwise youre going to have to get the image from a weird branch name